### PR TITLE
#285

### PR DESCRIPTION
**Pull request by dragospodariu96**
https://github.com/metasfresh/mf15/issues/285

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java [hunk 1]
```diff
@@ -885,7 +884,7 @@ private RecordsToOpen(
 			this.records = ImmutableList.copyOf(records);
 			this.windowIdString = StringUtils.trimBlankToNull(adWindowId);
 			this.target = target != null ? target : OpenTarget.GridView;
-			this.targetTab = targetTab != null ? targetTab : TargetTab.SAME_TAB;
+			this.targetTab = targetTab != null ? targetTab : target == OpenTarget.GridView ? TargetTab.SAME_TAB_OVERLAY : TargetTab.SAME_TAB;
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java:887]: avoid having logic here.
pls decide what kind of targetTab we need on the caller method,.

**Comment by TheBestPessimist** [comment]
@dragospodariu96 please 🍒 this PR to `intensive_care_uat` as well

**Comment by TheBestPessimist** [comment]
Another fix needed here:

In `de.metas.ui.web.process.adprocess.ADProcessPostProcessService#createResultAction` records to open might be null `final RecordsToOpen recordsToOpen = processExecutionResult.getRecordsToOpen();`.

This happens for example when running process `AllocatePayments`

image

**Review by TheBestPessimist — CHANGES_REQUESTED** [review]
please ignore this.

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java (hunk 1 above)
Comment (dragospodariu96) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java]: @teosarca then I would leave it as default in the builder to SAME_TAB unless set different and make sure whenever opentarget.gridview is used the targetTab gets passed as targettab.same_tab_overlay. WDYT?

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java [hunk 2]
```diff
@@ -493,7 +496,6 @@ public void setRecordToOpen(@Nullable final TableRecordReference record, final @
 					.record(record)
 					.adWindowId(adWindowId)
 					.target(target)
-					.targetTab(RecordsToOpen.TargetTab.SAME_TAB)
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java:496]: pls set the target tab too

**Pull request merged**